src/app/components/playback/playback_control.rs

```diff
@@ -149,6 +174,41 @@ impl PlaybackControl {
             .expect("error updating icon");
     }
 
+    fn update_shuffle(&self) {
+        let is_shuffled = self.model.state().playback.is_shuffled();
+        let playback_image = if is_shuffled && self.model.is_playing() {
+            "media-playlist-shuffle-symbolic"
+        } else {
+            "shuffle"
+        };
+        self.widget
+            .shuffle_button
+            .child()
+            .and_then(|child| child.downcast::<gtk::Image>().ok())
+            .map(|image| {
+                image.set_from_icon_name(Some(playback_image), image.icon_size());
+            })
+            .expect("error updating icon");
+    }
+
+
+    fn update_repeat(&self, mode: &RepeatMode) {
+        let playback_image = match mode {
+            RepeatMode::Song => "media-playlist-repeat-song-symbolic.symbolic",
```

Comment:
(not sure the .symbolic is needed)

Reply:
The .symbolic one looks better compared to normal media-playlist-repeat-song-symbolic in my opinion. It is larger and easier to understand

Reply:
I think there's no difference between `media-playlist-repeat-song-symbolic.symbolic` and `media-playlist-repeat-song-symbolic`, at least I dont see any! (I don't know the fine details of how the names are resolved but...)

More generally, try to only use icons from the standard system icons; https://flathub.org/apps/details/org.gnome.design.IconLibrary is a great tool for that! for instance, "shuffle" that you used above does not exist on my install

Reply:
Hmm ok did not think of that. I will have to try using only the standard icons

Reply:
thank you :)